DOC-4181 (#716)
corda/corda-docs-portal: content/en/platform/corda/5.0-dev-preview-2/getting-started/first-flow/first-flow.md

---
date: '2022-09-20'
title: "Your First Flow"
menu:
  corda-5-dev-preview2:
    parent: corda-5-dev-preview-start
    identifier: corda-5-dev-preview-flow
    weight: 5000
section_menu: corda-5-dev-preview2
---

The `MyFirstFlow` and `MyFirstResponderFlow` flows are basic flows that illustrate the main features of Corda 5 flows.
Many of the features will be familiar to those Developers who have written Corda 4 CorDapps. However, there are some important differences when using Corda 5:
* Services are injected on an as-needed basis.
* Flows are started via HTTP RPC, rather than a Java client sending Java classes over AMQP.
* Initiating and responder flows are linked with a protocol rather than class names.
* Singletons should be avoided in flow code because there is no guarantee that the same flow worker will continue to execute a flow after it has been check-pointed and restarted.

> It is also worth noting that, as of Developer Preview 2, there is no ledger service available to the flows. This means that you can not create tokens, transactions, or states yet.

Reviewer: ```suggestion
It is also worth noting that, as of Developer Preview 2, there is no ledger service available to the flows. This means that you cannot create tokens, transactions, or states yet.
```
